### DISCUSSION TITLE
Add colors

### DISCUSSION THREAD

**Discussion by iiasceri** [iBotPeaches/Apktool#3164] [Ideas]
At least get a green "apk built"

**Comment by iBotPeaches** [comment]
Do you mean in console output? Been fearful of colors since then this comes into play - https://no-color.org/

**Reply by iiasceri** [reply]
Yes but it mentions:NO_COLOR=1. In that case I vim ~/.zhrc and I am willing to add NO_COLOR=0.

**Reply by iiasceri** [reply]
And Color libraries supporting NO_COLOR to disable default color output are random projects I never touched or will touch so even adding bash no color eq 0 is pointless. In that case fork time (later)

**Reply by iBotPeaches** [reply]
So default to colors on and just have codebase respect `NO_COLOR`?

**Reply by iiasceri** [reply]
Yes each user decides if they want colors or no colors ideally they should be able to switch which projects are allowed to show colors and which are not allowed like "brew set nocolor apktool" or "brew set color apktook" brew inside will add path to tool and 0 or 1 to users liking

**Reply by iiasceri** [reply]
In this case I am running multiple commands including apktool b .apk and sometimes it fails and I need to scroll up and down. I have a simple 
.sh rebuild.sh that build uninstalls does install and launches main activity

**Reply by iBotPeaches** [reply]
okay never colored console Java output. So let me do some research if my formatters elegantly support this.

**Comment by iiasceri** [comment]
public class ColoredOutputExample {

    // ANSI escape codes for colors
    public static final String RESET = "\u001B[0m";
    public static final String BLACK = "\u001B[30m";
    public static final String RED = "\u001B[31m";
    public static final String GREEN = "\u001B[32m";
    public static final String YELLOW = "\u001B[33m";
    public static final String BLUE = "\u001B[34m";
    public static final String PURPLE = "\u001B[35m";
    public static final String CYAN = "\u001B[36m";
    public static final String WHITE = "\u001B[37m";

    public static void main(String[] args) {
        System.out.println(RED + "This text is red." + RESET);
        System.out.println(GREEN + "This text is green." + RESET);
        System.out.println(YELLOW + "This text is yellow." + RESET);
        System.out.println(BLUE + "This text is blue." + RESET);
        System.out.println(PURPLE + "This text is purple." + RESET);
        System.out.println(CYAN + "This text is cyan." + RESET);
        System.out.println(WHITE + "This text is white." + RESET);
    }
}

**Reply by iiasceri** [reply]
Note that the availability of ANSI escape sequences may vary depending on the terminal or console you are using. Some IDEs may not support colored output by default. However, you can use terminals like Command Prompt or Unix-based systems like Linux or macOS Terminal to see the colored output correctly.

**Reply by Krowne** [reply]
As they have said, depending on each terminal, the color output has to be implemented in a way. A variable could be added to choose the terminal type before compiling, so that when we compile we can choose the system type.

I have added the colors in the console messages from Windows 10 onwards. If @iBotPeaches is interested, he could share the codes of all the colors of the console for Windows 10 onwards. But it should be noted that these colors will only be displayed if the following entry exists in the Windows registry:

reg add "HKEY_CURRENT_USER\Console" /v VirtualTerminalLevel /t REG_DWORD /d 1 /f
